hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/query/QueryChecker.java

```diff
@@ -121,9 +121,9 @@ public void validateAccountBalances(
             throw new InsufficientBalanceException(xfersStatus, fee);
         }
 
-        // A super-user cannot use an alias. Sorry, Clark Kent. TODO Verify we have a test for this
-        final var accountOpt = payer.accountNumOrElse(AccountID.DEFAULT.accountNumOrThrow());
-        if (accountNumbers.isSuperuser(accountOpt)) {
+        // A super-user cannot use an alias. Sorry, Clark Kent.
+        final var accountNum = payer.accountNumOrElse(-1L);
```

Comment:
If one does not know the context, this looks pretty dangerous. Do we ensure somehow, that `-1` can never reference a superuser?

Reply:
-1 isn't a plausible account number, but I agree it looks dangerous. `payer.accountNumOrThrow()`?

Reply:
The code now checks `payer.hasAccountNum() && isSuperuser(accountNumOrThrow)` so we don't have to use -1. We also won't get the NPE behavior anymore.